src/common/threading.jl

```diff
@@ -6,51 +6,45 @@ Setup the number of threads used by DFTK's own threading (`n_DFTK`), by BLAS (`n
 and by FFTW (`n_fft`). This is independent from the number of Julia threads (`Threads.nthreads()`).
 DFTK and FFTW threading are upper bounded by `Threads.nthreads()`, but not BLAS,
 which uses its own threading system.
-By default, use 1 FFT thread, and `Threads.nthreads()` BLAS and DFTK threads
-Changing `n_DFTK` requires a restart of Julia.
+By default, use 1 FFT thread, and `Threads.nthreads()` BLAS and DFTK threads.
 """
 function setup_threading(; n_fft=1, n_blas=Threads.nthreads(), n_DFTK=nothing)
     if !isnothing(n_DFTK)
         set_DFTK_threads!(n_DFTK)
     end
-    n_DFTK = @load_preference("DFTK_threads", Threads.nthreads())
+    n_DFTK = get_DFTK_threads()
     FFTW.set_num_threads(n_fft)
     BLAS.set_num_threads(n_blas)
     mpi_master() && @info "Threading setup: " Threads.nthreads() n_DFTK n_fft n_blas
 end
 
 """
-Convenience function to disable all threading in DFTK and assert that Julia threading
-is off as well.
+Convenience function to disable all threading in DFTK.
 """
 function disable_threading()
     setup_threading(;n_fft=1, n_blas=1, n_DFTK=1)
 end
 
+# TODO: is a single write to an Int64 atomic?
+DFTK_threads = 0
```

Comment:
Should be const for sure.

But also this is global state. Do we really want this?

Reply:
Apparently according to Valentin the way to go is https://github.com/vchuravy/ScopedValues.jl, which will become part of base in 1.11. Can you look into this ?

Reply:
Funny how that's actually inspired from a Java API. 😄

I'm not sure that
```jl
with(DFTK_THREADS => 1) do
    # All computations here
end
```
is very user friendly however, especially since it's inconsistent with BLAS and FFTW threads.

Reply:
Making it `const` is a bad idea, the point is that it's actually mutable (and will be mutated by `set_DFTK_threads!`:
```
WARNING: redefinition of constant DFTK.DFTK_threads. This may fail, cause incorrect answers, or produce other errors.
```

Reply:
const does not mean that it's immutable if you make it a `Ref`. It most importantly it ensures the type does not change.

also `with(DFTK_THREADS => 1) do` can be wrapped.

Reply:
I talked to Valentin again, and the ideal thing to do would be to split work based on statically known block sizes (wherever we call `parallel_loop_over_range` at least), and let the Julia scheduler handle this.